spring-integration-core/src/main/java/org/springframework/integration/filter/MessageFilter.java

```diff
@@ -179,13 +180,16 @@ protected Object doHandleRequestMessage(Message<?> message) {
 	@Override
 	public Object postProcess(Message<?> message, Object result) {
 		if (result == null) {
-			MessageChannel channel = getDiscardChannel();
-			if (channel != null) {
-				this.messagingTemplate.send(channel, message);
+			MessageChannel channelToDiscard = getDiscardChannel();
+			if (channelToDiscard != null) {
+				this.messagingTemplate.send(channelToDiscard, message);
 			}
 			if (this.throwExceptionOnRejection) {
 				throw new MessageRejectedException(message, "message has been rejected in filter: " + this);
 			}
+			else if (channelToDiscard == null) {
+				logger.warn(LogMessage.format("The message [%s] has been rejected in filter: %s", message, this));
```

Comment:
WARN seems too strong; this might be the desired behavior; I would suggest DEBUG, perhaps with an option to increase the log level.

Reply:
I have no objection to requiring `nullChannel`, if it's only a 6.1 change.

Reply:
Yes, it is. Let me add such a sentence into doc and you are feel free to merge thereafter!